src/mscorlib/src/System/Runtime/Intrinsics/Vector128.cs

```diff
@@ -2,12 +2,56 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Diagnostics;
 using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
+using Internal.Runtime.CompilerServices;
 
 namespace System.Runtime.Intrinsics
 {
     [Intrinsic]
+    [DebuggerDisplay("{DisplayString,nq}")]
+    [DebuggerTypeProxy(typeof(Vector128DebugView<>))]
     [StructLayout(LayoutKind.Sequential, Size = 16)]
-    public struct Vector128<T> where T : struct {}
+    public struct Vector128<T> where T : struct
+    {
+        // These fields exist purely so debug view works (https://github.com/dotnet/coreclr/issues/15694)
+        private byte _00;
+        private byte _01;
+        private byte _02;
+        private byte _03;
+        private byte _04;
+        private byte _05;
+        private byte _06;
+        private byte _07;
+        private byte _08;
+        private byte _09;
+        private byte _0A;
+        private byte _0B;
+        private byte _0C;
+        private byte _0D;
+        private byte _0E;
+        private byte _0F;
+
+        private unsafe string DisplayString
+        {
+            get
+            {
+                // The IsPrimitive check ends up working for `bool`, `char`, `IntPtr`, and `UIntPtr`
```

Comment:
As per the comment, this ends up working for `bool`, `char`, `IntPtr`, and `UIntPtr` at debug time, but makes the subsequent code much simpler.

Given that this only works under a debugger, and users still can't do anything useful with such a type, I think this should be fine.